src/Listeners/DeleteBreadPermission.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace TCG\Voyager\Listeners;
+
+use TCG\Voyager\Events\BreadDeleted;
+use TCG\Voyager\Facades\Voyager;
+use TCG\Voyager\Models\Permission;
+use TCG\Voyager\Models\Role;
+
+class DeleteBreadPermission
+{
+    /**
+     * Create the event listener.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Delete Permission for a given BREAD.
+     *
+     * @param BreadDeleted $bread
+     *
+     * @return void
+     */
+    public function handle(BreadDeleted $bread)
+    {
+        if (config('voyager.bread.add_permission') && file_exists(base_path('routes/web.php'))) {
+            // Permission::generateFor(snake_case($bread->dataType->slug));
+            $role = Role::where('name', config('voyager.bread.default_role'))->firstOrFail();
+
+            // Get permission for added table
+            $permissions = Permission::where(['table_name' => $bread->dataType->name])->get()->pluck('id')->all();
+
+            // Assign permission to admin
+            $role->permissions()->detach($permissions);
```

Comment:
You're not actually deleting the permissions anywhere, just removing them from the default user.  Instead, they should be deleted and removed from all roles (ideally automatically per my comment above, or manually if soft-delete is used)